WebService/v1/Models/SimulationApiModel/SimulationApiModel.cs

```diff
@@ -188,6 +194,15 @@ public static SimulationApiModel FromServiceModel(Simulation value)
             }
 
             result.DeviceModels = SimulationDeviceModelRef.FromServiceModel(value.DeviceModels);
+            result.Statistics = SimulationStatistics.FromServiceModel(statistics);
+
+            if (result.Statistics != null)
+            {
+                DateTimeOffset now = DateTimeOffset.UtcNow;
+                TimeSpan duration = value.IsActiveNow ? now - value.StartTime.Value : value.StoppedTime.Value - value.StartTime.Value;
+                result.Statistics.AverageMessagesPerSecond = result.Statistics.TotalMessagesSent / duration.Seconds;
+            }
```

Comment:
this method is getting too long, and adding this code here increases the code complexity. Could you refactor this out.  

Also, make sure the division is not truncating decimals, e.g. `6/10` should return `0.6` and not `0` or `1`.